render/tests/test-event.js

```diff
@@ -848,6 +848,38 @@ o.spec("event", function() {
 			})
 		})
 	})
+	o("async function (event.redraw = false, await Promise, event.redraw = true)", function(done) {
+		var thenCB
+		var div = m("div", {onclick: async function (ev) {
+			// set event.redraw = false to prevent sync redraw
+			ev.redraw = false
+			await new Promise(function(resolve){thenCB = resolve})
+			// set event.redraw = true to enable async redraw
+			ev.redraw = true
+		}})
+		var e = $window.document.createEvent("MouseEvents")
+		e.initEvent("click", true, true)
+
+		render(root, div)
+		div.dom.dispatchEvent(e)
+
+		o(redraw.callCount).equals(0)
```

Comment:
Can you also assert `thenCB` is set? That being set also implies that the event listener was actually called.

Reply:
ditto